python/ml4ir/base/io/file_io.py

```diff
@@ -61,26 +64,17 @@ def read_df(
     Returns:
         pandas dataframe
     """
-    if infile.startswith("hdfs"):
-        # replace pydoop
-        # fp = hdfs.open(infile)
-        path_type = "HDFS"
+    if log_path:
+        log.info("Loading dataframe from path : {}".format(infile))
+
+    if infile.startswith(HDFS_PREFIX):
+        # NOTE: Move to fully spark dataframe based operations in the future
+        return spark_io.read_df(infile).toPandas()
```

Comment:
is this lazy? Does it scale or OOM?

Reply:
This is not lazy. The goal is to only use this for small files like vocabulary, etc which you really need in memory. For training data, we should use the lazy loading provided by TFRecordDataset.